test/unit/webgl/p5.RendererGL.js

```diff
@@ -23,6 +23,45 @@ suite('p5.RendererGL', function() {
     });
   });
 
+  suite('webglVersion', function() {
+    test('should return WEBGL2 by default', function() {
+      myp5.createCanvas(10, 10, myp5.WEBGL);
+      assert.equal(myp5.webglVersion, myp5.WEBGL2);
+    });
+
+    test('should return WEBGL1 after setAttributes', function() {
+      myp5.createCanvas(10, 10, myp5.WEBGL);
+      myp5.setAttributes({ version: 1 });
+      assert.equal(myp5.webglVersion, myp5.WEBGL);
+    });
+
+    suite('when WebGL2 is unavailable', function() {
+      let prevGetContext;
+      setup(function() {
+        prevGetContext = HTMLCanvasElement.prototype.getContext;
+        // Mock WebGL2 being unavailable
+        HTMLCanvasElement.prototype.getContext = function(type, attrs) {
+          if (type === 'webgl2') {
+            return undefined;
+          } else {
+            return prevGetContext.call(this, type, attrs);
+          }
+        };
+      });
+
+      teardown(function() {
+        // Put back the actual implementation
+        HTMLCanvasElement.prototype.getContext = prevGetContext;
+      });
+
+      test('should return WEBGL1', function() {
```

Comment:
I think you added this test twice?

Reply:
oops good catch, I left the `setAttributes({ version: 1})` in there by accident! I'll take that line out again so that it tests to make sure you still end up with webgl1 even when you don't request it if the browser doesn't support webgl2 (mocked in the `suite` this is nested in.)